fix: replace apostrophes with dashes

zikeji opened the pull request:
**Describe the change**
Fixes issue #281 by replacing apostrophes in instance name's with a dash.

**Checklist**
If you changed code:
- [+] Tests run locally and pass (`npm test`)
- [+] Code has correct format (`npm run format`)
   - Files I did not modify were formatted but those are not in this commit.

**Additional information**
This fixes the immediate issue but not sure it's the most thorough fix.

mKeRix commented:
Thanks for this fix - sorry it took so long until I merged it. A release with this included should follow shortly.

That said, I'm saving your comment on the original issue so that I can circle back to this. Thanks for the digging up the relevant code snippet in HA... we should of course follow the standard where possible.

zikeji commented:
> Thanks for this fix - sorry it took so long until I merged it. A release with this included should follow shortly.
> 
> That said, I'm saving your comment on the original issue so that I can circle back to this. Thanks for the digging up the relevant code snippet in HA... we should of course follow the standard where possible.

No problem! Thanks for getting to it.

I have a branch I just pushed up in my fork "ha-regex" where I did change the implementation in id.test and id.spec.ts to match HA's regex. But I also removed the underscore replacement - as underscores are allowed by HA in the topic. I wasn't sure if underscores cause a problem in room-assistant which is why I didn't pursue it any further.

https://github.com/zikeji/room-assistant/tree/ha-regex

The other issue I noticed was with foreign characters - I added a test in `id.spec.ts` (lines 18-20) for foreign characters. The issue is in both cases slugify removes (not replaces) foreign characters, and in that example `测试-房间` becomes simply `-`. The name '-' however passes HA's match (since their match is 1 or more characters). Which is fine, I guess, because HA doesn't care if it's only a single character. But the issue is when it's only foreign characters. Like `测试房间` would become an empty string.
